aws/adapter.go

```diff
@@ -399,3 +430,80 @@ func getTag(tags map[string]string, tagName string) (string, error) {
 	}
 	return "<missing tag>", ErrMissingTag
 }
+
+// Update list of ASGs that are updated with TargetGroup setting.
+func (a *Adapter) UpdateAutoScalingGroups(instancePrivateIps []string) error {
```

Comment:
Better name maybe: UpdateInstanceCaches

Reply:
This function updates both instance caches and list of known ASGs.
Cache is used to reduce number of calls to DescribeInstances.
I assume that instances are not taken out of ASG (strictly speaking it's wrong, but IMHO rarely used in production).
May be call it UpdateAutoScalingGroupsWithInstanceCache ?

Reply:
I am fine with this